Kwf/Component/Cache/Redis.php

```diff
@@ -0,0 +1,324 @@
+<?php
+class Kwf_Component_Cache_Redis extends Kwf_Component_Cache
+{
+    protected $_redis;
+    public function __construct()
+    {
+        $this->_redis = Kwf_Cache_Simple::getRedis();
+    }
+
+    public function save(Kwf_Component_Data $component, $contents, $renderer, $type, $value, $tag, $lifetime)
+    {
+        $key = self::_getCacheId($component->componentId, $renderer, $type, $value);
+
+        $this->_redis->sAdd('viewids:componentid:'.$component->componentId, $key);
+        $this->_redis->sAdd('viewids:dbid:'.$component->dbId, $key);
+        $this->_redis->sAdd('viewids:pagedbid:'.$component->getPageOrRoot()->dbId, $key);
+        $this->_redis->sAdd('viewids:cls:'.$component->componentClass, $key);
+        if ($tag) {
+            $this->_redis->sAdd('viewids:tag:'.$tag, $key);
+        }
+
+        $parts = preg_split('/([_\-])/', $component->getExpandedComponentId(), -1, PREG_SPLIT_DELIM_CAPTURE);
+        $id = '';
+        foreach ($parts as $part) {
+            $id .= $part;
+            if ($part != '-' && $part != '_' && $id != 'root') {
+                $this->_redis->sAdd('viewids:recexpandedid:'.$id, $key);
+            }
+        }
+
+        $cacheContent = array(
+            'contents' => $contents,
+            'expire' => is_null($lifetime) ? null : time()+$lifetime
+        );
+
+        if (is_null($lifetime)) {
+            //Set a TTL for view contents http://stackoverflow.com/questions/16370278/how-to-make-redis-choose-lru-eviction-policy-for-only-some-of-the-keys
+            $lifetime = 365*24*60*60;
+        }
+        $ret = $this->_redis->setEx($key, $lifetime, serialize($cacheContent));
+
+        return $ret;
+    }
+
+    public function load($componentId, $renderer='component', $type = 'component', $value = '')
+    {
+        $data = $this->loadWithMetadata($componentId, $renderer, $type, $value);
+        if ($data === null) return $data;
+        return $data['contents'];
+    }
+
+    public function loadWithMetadata($componentId, $renderer='component', $type = 'component', $value = '')
+    {
+        if ($componentId instanceof Kwf_Component_Data) {
+            $componentId = $componentId->componentId;
+        }
+        $key = self::_getCacheId($componentId, $renderer, $type, $value);
+        $ret = unserialize($this->_redis->get($key));
+        return $ret;
+    }
+
+    private function _deleteViewCache(array $updates, $dryRun)
+    {
+        $sets = array(
```

Comment:
I would favour $prefixes and handle "type" directly instead of "filterClientSide" (like it is already done in line 152)